src/Symfony/Component/HttpKernel/Tests/Debug/FileLinkFormatterTest.php

```diff
@@ -12,56 +12,61 @@
+use Symfony\Component\Routing\Router;
 
 class FileLinkFormatterTest extends TestCase
 {
-    public function testWhenNoFileLinkFormatAndNoRequest()
+    public function testWhenNoFileLinkFormat()
     {
         $sut = new FileLinkFormatter();
 
         $this->assertFalse($sut->format('/kernel/root/src/my/very/best/file.php', 3));
     }
 
-    public function testWhenFileLinkFormatAndNoRequest()
+    public function testWhenFileLinkFormat()
     {
         $file = __DIR__.DIRECTORY_SEPARATOR.'file.php';
 
-        $sut = new FileLinkFormatter('debug://open?url=file://%f&line=%l', new RequestStack());
+        $sut = new FileLinkFormatter('debug://open?url=file://%f&line=%l');
 
         $this->assertSame("debug://open?url=file://$file&line=3", $sut->format($file, 3));
     }
 
-    public function testWhenFileLinkFormatAndRequest()
+    public function testWhenFileLinkFormatAndRouter()
     {
         $file = __DIR__.DIRECTORY_SEPARATOR.'file.php';
         $baseDir = __DIR__;
-        $requestStack = new RequestStack();
-        $request = new Request();
-        $requestStack->push($request);
+        $router = $this->getRouter();
 
-        $sut = new FileLinkFormatter('debug://open?url=file://%f&line=%l', $requestStack, __DIR__, '/_profiler/open?file=%f&line=%l#line%l');
+        $sut = new FileLinkFormatter('debug://open?url=file://%f&line=%l', $router, $baseDir, '/_profiler/open?file=%f&line=%l#line%l');
 
         $this->assertSame("debug://open?url=file://$file&line=3", $sut->format($file, 3));
     }
 
-    public function testWhenNoFileLinkFormatAndRequest()
+    public function testWhenNoFileLinkFormatAndRouter()
     {
         $file = __DIR__.DIRECTORY_SEPARATOR.'file.php';
-        $requestStack = new RequestStack();
-        $request = new Request();
-        $requestStack->push($request);
+        $baseDir = __DIR__;
+        $router = $this->getRouter();
+
+        $sut = new FileLinkFormatter(null, $router, $baseDir, '?file=%f&line=%l#line%l');
 
-        $request->server->set('SERVER_NAME', 'www.example.org');
-        $request->server->set('SERVER_PORT', 80);
-        $request->server->set('SCRIPT_NAME', '/app.php');
-        $request->server->set('SCRIPT_FILENAME', '/web/app.php');
-        $request->server->set('REQUEST_URI', '/app.php/example');
+        $this->assertSame('/_profiler_customized?file=file.php&line=3#line3', $sut->format($file, 3));
+    }
+
+    private function getRouter()
+    {
+        $routeCollection = function () {
+            $routes = new RouteCollection();
+            $routes->add('_profiler_open_file', new Route('/_profiler_customized'));
 
-        $sut = new FileLinkFormatter(null, $requestStack, __DIR__, '/_profiler/open?file=%f&line=%l#line%l');
+            return $routes;
+        };
 
-        $this->assertSame('http://www.example.org/app.php/_profiler/open?file=file.php&line=3#line3', $sut->format($file, 3));
+        return new Router(new ClosureLoader(), $routeCollection);
```

Comment:
you could use `return new UrlGenerator($routecollection, new RequestContext())` (with `$routeCollection` being the RouteCollection itself rather than a closure returning it)